docs/adr/adr-013-automatic-gas-price-estimation.md

```diff
@@ -0,0 +1,65 @@
+# ADR #013: Automatic Gas Price Estimation
+
+## Changelog
+
+- 17.01.2024: Initial design
+
+## Status
+
+Proposed
+
+## Context
+
+[CIP 18](https://github.com/celestiaorg/CIPs/blob/main/cips/cip-18.md) defines an interface that a light node can use for transaction submission to retrieve an estimate of the gas price and amount of gas required to include their transaction on Celestia. This predominantly is to allow for greater dynamism during times of congestions where gas prices become greater than the global minimum.
+
+This ADR defines how the go light node will interact with the service as a client. Note this does not cover how the server will actually estimate the gas prices and gas required.
+
+## Decision
+
+Allow users of Celestia nodes to opt in to automatic gas price and gas usage estimation for every write method
+
+## Detailed Design
+
+### When to estimate the gas price
+
+Users can opt in to automatic price estimation by using the existing `TxConfig.isGasPriceSet`. Previously, if this was false, the `DefaultGasPrice` was used, now if a user isn't manually setting the gas price, the node will query it from the `GasEstimator` service.
+
+### When to estimate the gas required
+
+Depending on the message types submitted, the node will decide whether to also query for the amount of gas that is required to submit the transaction. In the case of the following messages, the gas can be reliably estimated using a simple linear model:
+
+- `MsgPayForBlob`
+
+For all other messages (or combination of messages), the node will call the `GasEstimator` service.
+
+### Setting guardrails
+
+A new field in the `TxConfig` will be introduced allowing for a user to  use the gas price estimation but setting a maximum gas price as a cap that the user is willing to pay. If the price returned by the estimator is greater than the cap, likely due to a price spike, an error will be returned informing the user that the gas price required is currently greater than the user is willing to pay
+
+### Defaults
+
+By default, the same consensus node that the node is connected with, should be used as the `GasEstimator`. During construction, a user can optionally provide a grpc address that can be used instead.
```

Comment:
Unclear what you mean by "a grpc address that can be used instead" 

Node has one entrypoint for tx submission and it's through a core node's gRPC. So you're saying the GasEstimator will also be available through exactly the same gRPC conn as other state-related queries we perform on the core node?

Reply:
My understanding is that we should provide new config field/flag for this service and try core GRPC if not specified.

Reply:
yes by default the consensus node will provide the `GasEstimator` service in the same gRPC connection. However users should be able to override that with a gRPC address of their own choosing which will be used just for gas/pricing estimation

Reply:
So an entirely new config field in the `state` config that can specify addr of estimator service. 

This should be included in ADR

Reply:
Yeah I wasn't sure how exactly it would be implemented. Should I just add that line

Reply:
Is there any chance that connection with this endpoint will be secured and we'll need a tls config or xToken to communicate with?